### PR TITLE
XW-2969 | removed suprfluous validation that caused optional categories not showing up

### PR DESCRIPTION
**Pull request by weronika**
https://wikia-inc.atlassian.net/browse/XW-2969

### CODE REVIEW

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 1]
```diff
@@ -618,12 +622,8 @@ private function extendSections( $curatedContent ) {
 			$section['items'] = $this->extendItemsWithImages( $section['items'] );
 			$section['items'] = $this->extendItemsWithType( $section['items'] );
 			return $section;
-		}, $curatedContent );
+		}, $data );
 
-		$data = array_values( array_filter( $data, function ( $section ) use ( $validator ) {
-			$error = isset( $section['featured'] ) ? false : $validator->validateSection( $section );
```
Comment (rogatty) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: It was introduced just 26 days ago, I suppose there was a reason :thinking:
@idradm please take a look.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 2]
```diff
@@ -605,6 +605,10 @@ private function sumUpQualityStats( $stats ) {
 	private function extendSections( $curatedContent ) {
 		$validator = new CuratedContentValidator();
 
+		$data = array_filter( $curatedContent, function ( $section ) {
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: 1. Should optional section be validated? - question from Adam

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 1 above)
Comment (weronika) [reply extensions/wikia/CuratedContent/CuratedContentController.class.php]: Yes, and I talked about the reason with Adam before introducing this change :)

**Comment by rogatty** [comment]
Let's add some tests as this logic gets pretty complicated.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 3]
```diff
@@ -604,26 +612,43 @@ private function sumUpQualityStats( $stats ) {
...
 			$section['items'] = $this->extendItemsWithType( $section['items'] );
-			return $section;
-		}, $curatedContent );
 
-		$data = array_values( array_filter( $data, function ( $section ) use ( $validator ) {
-			$error = isset( $section['featured'] ) ? false : $validator->validateSection( $section );
-			return empty( $error );
-		} ) );
+			if ( isset( $section['featured'] ) ) {
+				$section['items'] = array_filter( $section['items'], function ( $item ) use ( $validator ) {
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: you check if featured is set and then you operate on items, intentional?

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 4]
```diff
@@ -604,26 +612,43 @@ private function sumUpQualityStats( $stats ) {
...
+					$error = $validator->validateFeaturedItem( $item );
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: $section['items'] = array_filter($sections['items'], );

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 5]
```diff
@@ -352,25 +352,33 @@ public function getData() {
...
-				}, $this->communityDataService->getCurated() );
+				$curated = $this->communityDataService->getCurated();
+				if ( !empty( $curated ) ) {
+					$curated = array_map( function ( $section ) {
+						$section[ 'image_url' ] = CuratedContentHelper::findImageUrl( $section[ 'image_id' ] );
+						return $section;
+					}, $curated );
+
+					$curated[ 'curated' ] = 'true';
+					$sections = $curated;
```
Comment (rybmat) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: can we just assign result of the array_map function to `$sections` in L360?

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 6]
```diff
@@ -351,41 +351,56 @@ public function getData() {
...
 				}
 
 				$optional = $this->communityDataService->getOptional();
 				if ( !empty( $optional ) ) {
-					$curated[] = $optional;
-				}
+					$optional[ 'optional' ] = 'true';
+					$optional = $this->extendSectionAndItems( $optional );
+					$optional = $this->validateOptionalSection( $optional );
+					$sections[] = $optional;
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: should that be here?

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 7]
```diff
@@ -599,32 +614,78 @@ private function sumUpQualityStats( $stats ) {
...
+		} );
+
+		return $section;
+	}
+
+	/**
+	 * @param array $section optional categories section data
+	 * @return array input data with validated items only (optional categories).
+	 */
+	private function validateOptionalSection( $section ) {
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: private function validateOptionalSection( array $section ): array {}
please add types to functions

File: extensions/wikia/CuratedContent/CuratedContentController.class.php (hunk 6 above)
Comment (weronika) [reply extensions/wikia/CuratedContent/CuratedContentController.class.php]: no :)

**Pull request merged**